src/renderer_webgpu.cpp

```diff
@@ -0,0 +1,4808 @@
+			{
+				if (adapter.GetBackendType() == backendType)
+				{
+					backendAdapter = adapter;
+					break;
+				}
+			}
+
+			//BX_ASSERT(adapterIt != adapters.end());
+
+			WGPUDevice backendDevice = backendAdapter.CreateDevice();
+			DawnProcTable backendProcs = dawn_native::GetProcs();
+
+			using CreateSwapChain = DawnSwapChainImplementation (*)(wgpu::Device device, void* nwh);
+
+#if defined(DAWN_ENABLE_BACKEND_D3D12)
+			createSwapChain = CreateSwapChainD3D12;
+#elif defined(DAWN_ENABLE_BACKEND_METAL)
+			createSwapChain = CreateSwapChainMetal;
+#elif defined(DAWN_ENABLE_BACKEND_NULL)
+			createSwapChain = CreateSwapChainNull;
+#elif defined(DAWN_ENABLE_BACKEND_OPENGL)
+			createSwapChain = CreateSwapChainOpenGL;
+#elif defined(DAWN_ENABLE_BACKEND_VULKAN)
+			createSwapChain = CreateSwapChainVulkan;
+#endif
+
+			// Choose whether to use the backend procs and devices directly, or set up the wire.
+			WGPUDevice cDevice = backendDevice;
+			DawnProcTable procs = backendProcs;
+			dawnProcSetProcs(&procs);
+
+			m_device = wgpu::Device::Acquire(cDevice);
+#else
+			m_device = wgpu::Device(emscripten_webgpu_get_device());
+#endif
+
+			auto PrintDeviceError = [](WGPUErrorType errorType, const char* message, void*) {
+				BX_UNUSED(errorType);
+
+				if (s_ignoreError)
+				{
+					BX_TRACE("Device error: %s", message);
+				}
+				else
+				{
+					BX_CHECK(false, "Device error: %s", message);
+				}
+				s_ignoreError = false;
+			};
+
+			m_device.SetUncapturedErrorCallback(PrintDeviceError, nullptr);
+
+			if (!m_device)
+			{
+				BX_WARN(!m_device, "Unable to create WebGPU device.");
+				return false;
+			}
+
+			bool success = m_mainFrameBuffer.create(
+				  0
+				, g_platformData.nwh
+				, _init.resolution.width
+				, _init.resolution.height
+				, TextureFormat::Unknown
+				, TextureFormat::UnknownDepth
+				);
+			m_numWindows = 1;
+
+			if (!success)
+			{
+				return false;
+			}
+
+			m_queue = m_device.GetDefaultQueue();
+
+			m_cmd.init(m_queue);
+			//BGFX_FATAL(NULL != m_cmd.m_commandQueue, Fatal::UnableToInitialize, "Unable to create Metal device.");
+
+			for (uint8_t ii = 0; ii < WEBGPU_MAX_FRAMES_IN_FLIGHT; ++ii)
+			{
+				BX_TRACE("Create scratch buffer %d", ii);
+				m_scratchBuffers[ii].create(BGFX_CONFIG_MAX_DRAW_CALLS * 128);
+				m_bindStateCache[ii].create(); // (1024);
+			}
+
+			g_caps.supported |= (0
+				| BGFX_CAPS_ALPHA_TO_COVERAGE
+				| BGFX_CAPS_BLEND_INDEPENDENT
+				| BGFX_CAPS_FRAGMENT_DEPTH
+				| BGFX_CAPS_INDEX32
+				| BGFX_CAPS_INSTANCING
+			//	| BGFX_CAPS_OCCLUSION_QUERY
+				| BGFX_CAPS_SWAP_CHAIN
+				| BGFX_CAPS_TEXTURE_2D_ARRAY
+			//	| BGFX_CAPS_TEXTURE_3D
+				| BGFX_CAPS_TEXTURE_BLIT
+				| BGFX_CAPS_TEXTURE_COMPARE_ALL
+				| BGFX_CAPS_TEXTURE_COMPARE_LEQUAL
+				| BGFX_CAPS_TEXTURE_READ_BACK
+				| BGFX_CAPS_VERTEX_ATTRIB_HALF
+				| BGFX_CAPS_VERTEX_ATTRIB_UINT10
+				| BGFX_CAPS_COMPUTE
+				);
+
+			g_caps.limits.maxTextureSize   = 16384;
```

Comment:
I think the group agreed on 8k. It will be a limit exposed by the device. Ditto below.

Reply:
Fixed